WeatherMonitoring.Tests/Tests/ParsingTests.cs

```diff
@@ -0,0 +1,51 @@
+using Xunit;
+using FluentAssertions;
+using WeatherMonitoring.Parsers;
+using WeatherMonitoring.Parsers.Exceptions;
+
+public class ParserTests
+{
+    private readonly JsonParser _jsonParser = new();
+    private readonly XMLParser _xmlParser = new();
+
+    [Fact]
+    public void TryParse_ValidJson_ReturnsTrueAndData()
+    {
+        // Arrange
+        var input = @"{""Location"": ""Nablus"", ""Temperature"": 32, ""Humidity"": 40}";
+        // Act
+        var data = _jsonParser.Parse(input);
+
+        // Assert
+        data.Should().NotBeNull();
+        data!.Location.Should().Be("Nablus");
+        data.Temperature.Should().Be(32);
+        data.Humidity.Should().Be(40);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("not json")]
+    [InlineData(@"{ ""Location"": ""Nablus"" }")] // missing fields
+    public void TryParse_InvalidJson_ReturnsFalse(string input)
```

Comment:
Should be either Testing for Throws or we need to change the test to actually check for the False value

```suggestion
    public void TryParse_InvalidJson_ThrowsException(string input)
```